packages/domain-picker/test/suggestion-item.tsx

```diff
@@ -15,293 +16,76 @@ import { render, fireEvent, screen } from '@testing-library/react';
-					railcarId: 'id2',
-					result: 'example2.com',
-					trainTracksType: 'render',
-					uiPosition: 2,
-				} )
-			);
+			expect( updatedProps.onRender ).toHaveBeenCalledTimes( 1 );
 		} );
 
-		[
-			{ prop: 'railcarId', value1: 'id1', value2: 'id2' },
-			{ prop: 'isRecommended', value1: true, value2: false },
-			{ prop: 'categorySlug', value1: null, value2: 'test_category' },
-			{ prop: 'uiPosition', value1: 1, value2: 2 },
-			{
-				prop: 'suggestion',
-				value1: { ...testSuggestion, domain_name: 'example1.com' },
-				value2: { ...testSuggestion, domain_name: 'example2.com' },
-			},
-		].forEach( ( { prop, value1, value2 } ) => {
-			it( `doesn't send render event when only "${ prop }" changes`, async () => {
-				// Delay import so we have time to load configData in `beforeAll`
-				const { default: SuggestionItem } = await import( '../suggestion-item' );
-
-				const recordAnalytics = jest.fn();
-
-				const props = {
-					suggestion: testSuggestion,
-					onSelect: jest.fn(),
-					railcarId: 'id',
-					recordAnalytics,
-					uiPosition: 113,
-					[ prop ]: value1,
-				};
-
-				const { rerender } = render( <SuggestionItem { ...props } /> );
-
-				expect( recordAnalytics ).toHaveBeenCalledWith(
-					expect.objectContaining( {
-						railcarId: props.railcarId,
-						result: expect.stringContaining( props.suggestion.domain_name ),
-						trainTracksType: 'render',
-						uiPosition: props.uiPosition,
-					} )
-				);
-
-				recordAnalytics.mockClear();
-				props[ prop ] = value2;
-
-				rerender( <SuggestionItem { ...props } /> );
-
-				expect( recordAnalytics ).not.toHaveBeenCalled();
-			} );
+		it( 'should send a render event when domain name and railcarId changes', async () => {
+			const { rerender } = renderComponent();
+			const updatedProps = { ...MOCK_PROPS, domain: 'example1.com', railcarId: 'id1' };
+			rerender( <SuggestionItem { ...updatedProps } /> );
+
+			expect( updatedProps.onRender ).toHaveBeenCalledTimes( 2 );
 		} );
 	} );
 
 	describe( 'interact event', () => {
-		it( 'sends interact event when selected', async () => {
-			// Delay import so we have time to load configData in `beforeAll`
-			const { default: SuggestionItem } = await import( '../suggestion-item' );
-
-			const recordAnalytics = jest.fn();
-			const railcarId = 'id';
-
-			const { getByLabelText } = render(
-				<SuggestionItem
-					isSelected={ false }
-					suggestion={ testSuggestion }
-					onSelect={ jest.fn() }
-					railcarId={ railcarId }
-					recordAnalytics={ recordAnalytics }
-					uiPosition={ 113 }
-				/>
-			);
-
-			fireEvent.click(
-				// Use function matcher to deal with the label containing <spans>
-				getByLabelText( ( content, element ) => element.textContent === 'example.com' )
-			);
-
-			expect( recordAnalytics ).toHaveBeenCalledWith( {
-				trainTracksType: 'interact',
-				action: 'domain_selected',
-				railcarId,
-			} );
-		} );
+		it( 'should send an interact event when selected', async () => {
```

Comment:
How do we deal with different item types? Since this is a unit test, should we write tests for all types?

Reply:
We currently have three options: `button`, `row` & `radio` and I think that we should cover all scenarios (either with unit or integration tests). I think that we can include it in the A/C of #51175. WDYT?

Reply:
Sounds good to me, let's include them in #51175

Reply:
I've updated the A/C.